sdk/trace/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorTest.java

```diff
@@ -126,6 +126,24 @@ void builderInvalidConfig() {
         .hasMessage("maxQueueSize must be positive.");
   }
 
+  @Test
+  void builderAdjustMaxBatchSize() {
+    SpanExporter dummyExporter = new CompletableSpanExporter();
+
+    BatchSpanProcessorBuilder smallQueueBuilder =
+        BatchSpanProcessor.builder(dummyExporter).setMaxQueueSize(1).setMaxExportBatchSize(1000);
+    smallQueueBuilder.build();
+
+    assertThat(smallQueueBuilder.getMaxExportBatchSize()).isEqualTo(1);
+
+    BatchSpanProcessorBuilder largeBatchBuilder =
+        BatchSpanProcessor.builder(dummyExporter).setMaxQueueSize(513).setMaxExportBatchSize(1000);
+    largeBatchBuilder.build();
+
+    assertThat(largeBatchBuilder.getMaxExportBatchSize())
+        .isEqualTo(BatchSpanProcessorBuilder.DEFAULT_MAX_EXPORT_BATCH_SIZE);
```

Comment:
could we split this into two separate tests? aka one where it selects `maxQueueSize` and another where it selects `DEFAULT_MAX_EXPORT_BATCH_SIZE`